modules/assim.sequential/R/Multi_Site_Constructors.R

```diff
@@ -24,20 +24,20 @@ Contruct.Pf <- function(site.ids, var.names, X, localization.FUN=NULL, t=1, bloc
     #let's find out where this cov (for the current site needs to go in the main cov matrix)
     pos.in.matrix <- which(attr(X,"Site") %in% site)
    #foreach site let's get the Xs
-    pf.matrix [pos.in.matrix, pos.in.matrix] <- cov( X [, pos.in.matrix] ,use="complete.obs")
+    pf.matrix [pos.in.matrix, pos.in.matrix] <- stats::cov( X [, pos.in.matrix] ,use="complete.obs")
   }
   
   # This is where we estimate the cov between state variables of different sites
   #I put this into a sperate loop so we can have more control over it
-  site.cov.orders <- expand.grid(site.ids,site.ids) %>%
-                        filter( Var1 != Var2)
+  site.combs <- expand.grid(site.ids,site.ids)
+  site.cov.orders <- site.combs[which(site.combs$Var1 != site.combs$Var2),]
```

Comment:
Not sure why you got rid of the `filter` here. I'd recommend putting it back because the original code was cleaner and easier to understand